runtime/modules/render_runtime/data.go

```diff
@@ -54,3 +56,38 @@ func DataSeriesFromStarlark(list *starlark.List) ([][2]float64, error) {
 
 	return result, nil
 }
+
+func WeightsFromStarlark(list *starlark.List) ([]float64, error) {
```

Comment:
nit: Instead of calling this weights, it might be worth calling this something more generic so it can be reused as a list of floats.